src/widgets/widget-model.js

```diff
@@ -44,6 +67,83 @@ module.exports = cdb.core.Model.extend({
     this.stopListening();
   },
 
+  isAutoStyleEnabled: function (autoStyle) {
+    var styles = this.get('style');
+
+    if (!styles && (this.get('type') === 'category' || this.get('type') === 'histogram')) return true;
+    return styles && styles.auto_style && styles.auto_style.allowed;
+  },
+
+  getWidgetColor: function () {
+    var styles = this.get('style');
+
+    return styles && styles.widget_style &&
+          styles.widget_style.definition &&
+          styles.widget_style.definition.color &&
+          styles.widget_style.definition.color.fixed;
+  },
+
+  getColor: function (name) {
+    if (this.isAutoStyleEnabled() && this.isAutoStyle()) {
+      return this.autoStyler.colors.getColorByCategory(name);
+    } else {
+      return this.getWidgetColor();
+    }
+  },
+
+  isAutoStyle: function () {
+    return this.get('autoStyle');
+  },
+
+  autoStyle: function () {
+    if (!this.isAutoStyleEnabled()) return;
+
+    var layer = this.dataviewModel.layer;
+
+    if (!layer.get('initialStyle')) {
+      var initialStyle = layer.get('cartocss');
+      if (!initialStyle && layer.get('meta')) {
+        initialStyle = layer.get('meta').cartocss;
+      }
+      layer.set('initialStyle', initialStyle);
+    }
+
+    var style = this.autoStyler.getStyle();
+    layer.set('cartocss', style);
+    this.set('autoStyle', true);
+    // this.getAutoStyle();
+    return true;
+  },
+
+  reapplyAutoStyle: function () {
+    var style = this.autoStyler.getStyle();
+    this.dataviewModel.layer.set('cartocss', style);
+    this.set('autoStyle', true);
+  },
+
+  cancelAutoStyle: function (noRestore) {
```

Comment:
As @alonsogarciapablo told me this morning, from a reader point of view, it's better passing an options objects with keys:
```
cancelAutoStyle: function (opts) {
   if (opts.noRestore === false) {
   ...
   }
}
```